distribution/client/src/lib/storage.ts

```diff
@@ -98,6 +103,18 @@ export default class Storage {
         });
 
       }));
+      retArray.push(new Promise((resolve, reject) => {
+
+        fs.unlink(`${pluginPath}/${plugin}.jpi`, (err) => {
+          if (err) {
+            logger.debug(`${pluginPath}/${plugin}.jpi was not found.`);
```

Comment:
Do you want to reject here?

Reply:
@jvz not *really*. We should actually reject or at least in case it's something else than a file not found, but it's a known issue we are going to tackle very soon https://issues.jenkins-ci.org/browse/JENKINS-54207

Reply:
Since there's not a `stat` operation happening before this, an error might be non-fatal (i.e. "not found". I don't think rejecting would make sense on this tbh